File: mediator-simple-java-demo/src/main/java/de/codecentric/cm4sb/demo/mediator/service/MovieService.java
package de.codecentric.cm4sb.demo.mediator.service;

import de.codecentric.cm4sb.demo.mediator.domain.Movie;
import de.codecentric.cm4sb.demo.mediator.repository.MovieRepository;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

@Service
public class MovieService {
    private final MovieRepository movieRepository;

    public MovieService(MovieRepository movieRepository) {
        this.movieRepository = movieRepository;
    }

    public Movie getMovie() {
        return this.getRecommendedMovie();
    }

    private List<Movie> getMovies() {
        List<Movie> movies = new ArrayList<>();
        movieRepository.findAll().forEach(movies::add);
        return movies;
    }

    private Movie getRecommendedMovie() {
        List<Movie> movies = getMovies();
        return movies.get(new Random().nextInt(movies.size()));
    }
}

File: mediator-simple-kotlin-demo/src/main/kotlin/de/codecentric/cm4sb/demo/mediator/service/MovieService.kt
package de.codecentric.cm4sb.demo.mediator.service

import de.codecentric.cm4sb.demo.mediator.domain.Movie
import de.codecentric.cm4sb.demo.mediator.repository.MovieRepository
import io.vavr.control.Try
import org.springframework.stereotype.Service
import java.util.*

@Service
class MovieService(val movieRepository: MovieRepository) {
    private val fallbackMovie: Movie = Movie("Titanic", "3h 15m")

    fun getMovies(): List<Movie> = movieRepository.findAll().toList()

    fun getMovie(): Movie = Try.of(this::getRecommendedMovie).getOrElse(fallbackMovie)

    fun getRecommendedMovie(): Movie {
        val movies = getMovies()
        return movies[Random().nextInt(movies.size)]
    }

}
